app/src/main/java/com/example/sodastreamprototyping/TensorFlowAPI.kt

```diff
@@ -1,2 +1,57 @@
-package com.example.sodastreamprototyping.ui.theme
+package com.example.sodastreamprototyping
 
+import android.content.Context
+import org.tensorflow.lite.Interpreter
+import java.nio.ByteBuffer
+import java.nio.MappedByteBuffer
+import java.nio.channels.FileChannel
+import java.io.FileInputStream
+
+class DrinkGenerator(context: Context){
+
+    private val interpreter: Interpreter
+
+    init {
+        interpreter = Interpreter(loadModelFile(context))
+    }
+
+    private fun loadModelFile(context: Context): MappedByteBuffer{
+        val fileDescriptor = context.assets.openFd("models/model.tflite")
+        val inputStream = FileInputStream(fileDescriptor.fileDescriptor)
+        val fileChannel = inputStream.channel
+        val startOffset = fileDescriptor.startOffset
+        val declaredLength = fileDescriptor.declaredLength
+        return fileChannel.map(FileChannel.MapMode.READ_ONLY, startOffset, declaredLength)
+    }
+
+    // soda represents an integer index of the base soda
+    // flavors is an array with length of 18 that represents the concentration of each flavor in the drink
+    fun generateDrink(soda : Int, flavors: FloatArray) : List<String> {
```

Comment:
I think that for the purpose of readability, we should have the AI take in an array of ints instead, and then divide each value by 3. Other portions of the app can shouldn't need to worry that each pump is 0.33 etc.

Reply:
Remember that I trained the AI do use the soda value divided by 3 too.